src/NewTools-Debugger-Tests/StDebuggerTest.class.st

```diff
@@ -752,6 +759,56 @@ StDebuggerTest >> testNewDebuggerContextFor [
 	self assert: debuggerContext context identicalTo: ctx
 ]
 
+{ #category : 'tests - context inspector' }
+StDebuggerTest >> testPreserveLastKnownExtent [
+
+	| newDbgExtent lastUserExtent |
+
+	dbg := (self debuggerOn: session) open.
+	"First we save the user extent (test running should not modify user changes)"
+	lastUserExtent := self settings lastKnownExtent.
+	newDbgExtent := 100 @ 200.
+	"Programmatically change the extent of debugger window"
+	dbg window window extent: newDbgExtent.
+	"New extent is saved when the window is closed"
+	dbg close.
+	self tearDown.
+
+	"Now we open the debugger again"
+	self setUp.
+	dbg := (self debuggerOn: session) open.
+	"Check the debugger window is set to the new extent"
+	self assert: dbg window window extent equals: newDbgExtent.
+	"Set back the last extent regardless of this test"
+	dbg window window extent: lastUserExtent.
+
+]
```

Comment:
this test is failing

Reply:
Which Pharo version do you use?

Using the latest Pharo 13.0.0 here, and it works:
Build information: Pharo-13.0.0+SNAPSHOT.build.124.sha.d1dbcc0440c7d27e9a05e6ea2a659f5ddb41a484 (64 Bit)

![Screenshot 2024-07-15 at 15 53 55](https://github.com/user-attachments/assets/f7233034-2500-4fe5-aed8-61540f53188b)

Reply:
the last P13 build (look at the CI)

Reply:
I see, I don't understand why is it failing in the CI.

I would rewrite the test following your suggestions, although I think it doesn't reflect the actual user interaction.